### PR TITLE
Update tifffile>=2023.4.12 in requirements.txt

### LINKED ISSUE #62: GeoTIFFs not being loaded with tifffile 2022.4.26
When using `tifffile` version 2022.4.26 (as specified in the requirements.txt), all GeoTIFF files I tried to load read as NODATA (0 in below example). When I manually update `tifffile` to 2023.4.12 or newer, the data is being loaded properly. I tried loading these files with `tifffile` directly and the beahvior was the same for older versions, meaning this is not an issue with this library's code. Additionally, this issue only manifests itself when loading as zarr, if loading to a numpy array, the data is loaded correctly.

The code I used for testing:
```py
from geotiff import GeoTiff
import numpy as np

tiff = GeoTiff('FAA_UTM18N_NAD83.tif')

data = tiff.read()

print(np.mean(data))
```

|`tifffile` version|output|
|---|---|
|<=2023.3.21|0|
|>=2023.4.12|202.33013993598698|

The used GeoTIFF file can be found [here](https://www.sciencebase.gov/catalog/file/get/53f5a87ae4b09d12e0e8547b?f=__disk__78%2Fc8%2F28%2F78c82825988e79ba1608639fd9d87e78104e5bce
).

### PR DESCRIPTION
**Pull request by doyled-it**
Closes #62

I ran the code in #62 with this `tifffile` version installed and I received the same output.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-tifffile>=2021.7.2,<2022.4.28
+tifffile>=2023.4.12
```
Comment (Zeitsperre) [inline requirements.txt:1]: Does opening the pin for `>=2021.7.2` do the job, or are there regressions that were introduced in version `2022.4.28`?

Comment (Zeitsperre) [reply requirements.txt:1]: Never mind, I see the issue now. Perhaps we can mark geotiff to skip `tiffile==2022.4.28`?

Comment (Zeitsperre) [reply requirements.txt:1]:
```suggestion
tifffile>=2021.7.4,!=2022.4.28
```

**Comment by Zeitsperre** [comment]
The bad version appears to have been `tifffile@v2022.4.26`. I'll need to go through the changelogs here and there to figure out what we should be pinning.

I'm tempted to try testing against several versions of `tiffile` to see exactly which versions should be avoided.

Feel free to ignore the Python3.9 failure (addressed in #65).

**Comment by doyled-it** [comment]
> The bad version appears to have been `tifffile@v2022.4.26`. I'll need to go through the changelogs here and there to figure out what we should be pinning.
> 
> I'm tempted to try testing against several versions of `tiffile` to see exactly which versions should be avoided.
> 
> Feel free to ignore the Python3.9 failure (will be addressed in #65).

I was just told by a colleague @mdotter-mitre that even version `2023.3.21` of `tifffile` produces the same error.

**Comment by mdotter-mitre** [comment]
It looks like from [`this issue`](https://github.com/KipCrossing/geotiff/issues/62#issue-1824578206), any version below and including `tifffile<=2023.3.21` will result on the NODATA error, not isolated in `tifffile==2022.4.26`. Recommend setting tifffile>=2023.4.12`.

**Comment by Zeitsperre** [comment]
@doyled-it @mdotter-mitre 

Thanks for pointing this out.

I'm reluctant to break backwards compatibility (recent versions of `tifffile` drop both Python3.7 and 3.8), so maybe we can make use of [PEP 508](https://peps.python.org/pep-0508/)'s environment markers, i.e.
```
tifffile>=2021.7.4,<the.version.where.it.all.went.wrong; python_version>=3.7,<3.9
tifffile>=2023.4.12; python_version>=3.9
```

Do we know for certain that the regression was introduced in `tifffile==2022.4.26`, or did it happen earlier?

**Comment by KipCrossing** [comment]
Thanks for looking into this. I'll check it out when I'm up (Aus time).

@Zeitsperre I'll give you ownership of this repo and the PyPI so you can update at will. 

---

Also, looks like mypy is failing in version in 3.9.

**Comment by Zeitsperre** [comment]
@KipCrossing Thanks so much! My PyPI username is the same as on here. I'll do my best to not make any egregious packaging errors!

AFAIK, the mypy error is addressed here: https://github.com/KipCrossing/geotiff/pull/65/commits/fa6598ca84be872f2e89e24bd6526aa18db8a70b (on further review, I should clean that up).

**Comment by KipCrossing** [comment]
@doyled-it do you mind pulling the latest main into this branch?

**Comment by KipCrossing** [comment]
@Zeitsperre I've added you to PyPI. Is there any additional permissions that you need here on GitHub so, if I'm AFK for a while you can do everything? 

---

In general, it would be good to get more contributes on this project. The UI is pretty ugly and could use an overhaul. I'm not really in the GIS space these days so I don't have any strong opinions, tho. 

(sorry for dumping this in the PR haha)

**Comment by doyled-it** [comment]
> @doyled-it do you mind pulling the latest main into this branch?

I'm kinda new to GitHub PRs and forking as we mostly work with GitLab MRs, so I hope I did that correct here.

**Comment by KipCrossing** [comment]
> @KipCrossing I'm kinda new to GitHub PRs and forking as we mostly work with GitLab MRs, so I hope I did that correct here.

Perfect! I'll merge to main and fix the typing problem before deploying to PyPI

**Pull request merged**